src/LogicBlock/SMTLibLogicModel.cpp

```diff
@@ -0,0 +1,131 @@
+
+#include "LogicBlock/SMTLibLogicModel.hpp"
+
+#include <regex>
+
+std::string smtliblogic::SMTLibLogicModel::getTermString(const LogicTerm& a) {
+    std::stringstream ss(modelString);
+    std::string       line;
+
+    const auto  regex = std::regex("\\(define-fun (.*) \\(\\) (.*) (.*)\\)");
+    std::smatch m;
+    while (std::getline(ss, line)) {
+        if (std::regex_search(line, m, regex)) {
+            if (m.size() == 4) {
+                if (m[1] == a.getName()) {
+                    return line;
+                }
+            }
+        }
+    }
+    return {};
+}
+std::string smtliblogic::SMTLibLogicModel::getTermValue(const logicbase::LogicTerm& a) {
+    std::string const termString = getTermString(a);
+
+    const auto  regex = std::regex(R"(\(define-fun (.*) \(\) (.*) (.*)\))");
+    std::smatch m;
+
+    if (std::regex_search(termString, m, regex)) {
+        if (m.size() == 4) {
+            return m[3];
+        }
+    }
+
+    return {};
+}
+int smtliblogic::SMTLibLogicModel::getIntValue(const logicbase::LogicTerm& a, logicbase::LogicBlock* lb) {
+    std::string const value = getTermValue(a);
+    const auto        type  = getTermType(a);
+    if (type != "Int") {
+        throw std::runtime_error("SMTLibLogicModel::getIntValue: type mismatch");
```

Comment:
This is just a very minor nit on two points: If an exception is thrown, it always contains the information from where it was thrown. So the `SMTLibLogicModel::getIntValue` part in the message is kind of redundant.
Second, it's always good to include available information about the error. In the case here, what was the expected type and what was received. Something like
```c++
        throw std::runtime_error("type mismatch. Expected int, got" + type");
```
Please also adapt in the error messages below.